---
layout: post
title: 常用命令记录
key: 20180505
tags: command
---

```bash
for VARIABLE in 1 2 3 4 5 .. N
do
	command1
	command2
	commandN
don
```
<!--more-->

OR
```bash
for VARIABLE in file1 file2 file3
do
	command1 on $VARIABLE
	command2
	commandN
done
```
OR
```bash
for OUTPUT in $(Linux-Or-Unix-Command-Here)
do
	command1 on $OUTPUT
	command2 on $OUTPUT
	commandN
done
```
Examples:
```bash
for i in 1 2 5 6 7 8
do
cp /local/wen/FMD4/L$i/*vcf* ./
done
```
```bash
for i in `ls *.sam`
do
    echo "predict_graphs.py $i" >> command.predict_graphs.list
done

for i in `seq 1 100`  
do  
    echo $i  
done  
```
```bash
for i in {1..5}
```

`{START..END..INCREMENT} `
```bash
for i in {0..10..2}
```
```bash
for i in $(seq 1 2 20)
```
Three-expression bash for loops syntax:
```bash
for (( EXP1; EXP2; EXP3 ))


do
	command1
	command2
	command3
done
```
```bash
#!/bin/bash
for (( c=1; c<=5; c++ ))
do  
   echo "Welcome $c times"
done
```
see [more](https://www.cyberciti.biz/faq/bash-for-loop/)

### awk
```
awk '{if($3 == 100||$3 < 50){print $3}}' DHS1085091 | wc -l >>judge
awk  'NR==FNR{a[$0]}NR>FNR{ if($4 in a)    print $0}' DHS DHS_hg19.pos >_DHS
awk  'NR==FNR{a[$0]}NR>FNR{ if($4 in a)    print $0}' DHS_use DHS_hg19.pos_sorted > DHS_use_hg19.pos_sorted
awk  'NR==FNR{a[$0]}NR>FNR{ if($4 in a)    print $0}' AR AR_hg19.pos_sorted > AR_use_hg19.pos_sorted
bedtools intersect -a AR_use_hg19.pos_sorted -b hg19_seg_dup.bed_sort -wa
根据encode筛选后：comm -23 AR rm_AR > final_AR
awk  'NR==FNR{a[$0]}NR>FNR{ if($2 in a)    print $0}' final_AR 01final.map > final_map
awk  'NR==FNR{a[$0]}NR>FNR{ if($5 in a)    print $0}' dhs DHS_sites_n.bed > dhs_count_pos
cat dhs_count_pos | awk '{sum += $2} END {print sum}'
awk '{if ($4 ~/T/ && $5 ~/A/) print}' mutation_CpG
```

### bedtools
```
bedtools getfasta -fi /local/wenhm/annotation/hg19.fa -bed somatic_mutations_GC_region.bed -fo GC
```
输出文件名是GC，fasta序列是这样格式的：
>chr10:3183199-3184198
cacagctcactgcagctttgaactcctgggctcaagggatcctcccacaacagcctcctgagtagctgggcctacaggtgtgtgccaccacgcctgactatttttacagtttttgtagaggcagggtctagctttgttgcccaggctggtctcaaacacctggcttcaagtgatcctcctgccttcaaagtgctgagcttacaggcatgagATTTTCTTTTCAATTAACCAAAACTAAAAAGCACAAGGAAGCACAATTTTCAAACAGTTGGCTTAGATTACAGGGTTGAAGGGAGAGGTGGTGGGGAGGGAATCTTAACACTTCCAGCACCACACTGCTTAACTTGTTCCAATGCACATGTGGCATTTTTAAAATCTCGTGAAGTAGAAAGAGGTTATTTCTATTATAAAGGCAAATGAGCAAATGAACATTTATTTAAATGTCTCACATCAGAATATGGAAAGAAAGGAAACTGAAGTGAGTTTTATAAGAATCCCTTAAGAAAACACACACTCTCTGATAAAGCACATTTCTAGGAACAGGGCCAATTTCCTCTGCCTTCCACTGTGAACGGTAACATCACATTCAAGCTGATTCTAGAACACATCCACCTCCGACAGAAGGCGGGGCAGACAATGATGGGAGGGTCTGCAGCACAGACACAGGCTCCTTGAAGACGCCCGTGGTGAGATCAGCCCCAGGTCGCCGCTTCCACAGACAGGCCTTTTCCTAAAGTGGATGAAGGGAGTCAAATAGGCCAGGTGGGAAATCAAGGAAGGAGTGGGCGTGTTCCCAGGGGCTGTGGTGGGAGAAGTCACGACCATCTCGTGTGGGTCCTCCAGGAGAAACCCAGCGCCACTCGGGTGGCTGGGGCGCCGTCAGCCGGGACTAGCGAGGCTGGAGTCAGACCCTGGGGGACGGAGTCTCCACCAGCACCGGCAGGGATGGCGTTCCATCCGTCCACACATTATCCTTCCTGAGAAATTTCTATTACTTGT

>chr10:3292694-3293693
cctatgagaagagcaggcagacacccccgtttataaatgaggaaaTGAgctggacacagtggctcactcctgtgatcccagcactttgggaggccgaggcaggtggatcacctgaggtcaggagtttgagaccagcctggccaacatggcaaaaccctatctctactaaaaatacaaaagttagctggacgtggtggcacgtgcctgtagttccagctacttgggaggctgaggcaggagaatcagttgaacccaggaggcggagattgcagtgagccgagattgtgctactgcactccagcctgggtgacgagagcgaaaactccgtctcaaaaaaCAaataataataataataataaATGAGGAAACAGTTGCTCAGCACTGTTGGGGCATGGCCAAGAGCACACAGAGATGGAGCTGGGCTTGTTTGGAGGTTTCTGGTGATAAGTCCCAGCTCTTGTACCACCCCATCCCTACAGGACTCCCAGGTTTTCTGAACGGTATAAATGATCTCACAGTCAGGAAGCAGGACACCATAGCTCACCTGAGCAACCGGAGGGCAGGCAAGCAGTTAGCCAGTTTGCTGTCACTCTCTTAGCCCTCTGGTTCTGGCCTTTGTGAGTGAATGTGAATAGGAAGAACATGAAAGGAAAATAATCCTAATATCCCATAGTAGTTCCCCATCCAAGCACCCTTTCTCTTCAAACCTCCTAAAGCTTCGACAACCACCTGGTGCTCATGATCTCCTCTCTCATCTCCTCAGAAGACACGGTTCTTGGGTCTTCTCTCTCTTATTCTTCAGAATGAGTCTTCCCTCCAGAAACTCCTCTGAGATTTGGCTGTGGTCTCCCACGCCCTGTCAGGCAGCGACTGCAGGCTTCTGTGGTTCTCGTTTCCAAACCTCAAACCATGAGAGACATGGAGGCTCTTCGAAATGATGGTGACACTGCTCTCCTGGGGAGGGCTGGACATGGATCGTCCTTTCCCAGCTCCGCTTAA

```
ls 路径/* 能把该路径下的文件的绝对路径显示出来
```

### 计算genome size的方法找到啦
```bash
samtools faidx genome.fa
awk {'print $1,"\t",$2'} genome.fa.fai >genome.size
```

超级方便啊！！！

### 下载hg19基因组文件
```bash
wget http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.2bit
wget http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/twoBitToFa
```
## converting a hg19 refGene.txt to GTF
```bash
rsync -a -P rsync://hgdownload.soe.ucsc.edu/goldenPath/hg19/database/refGene.txt.gz ./
gzip -d refGene.txt.gz
cut -f 2- refGene.txt > refGene
genePredToGtf file refGene.input hg19refGene.gtf
cat hg19refGene.gtf  | sort -k1,1 -k4,4n > hg19refGene.gtf.sorted
```
## 查找左右py脚本
```bash
find -type f -name '*.py'
```
## 时间排序
```bash
ls -lhtr
```
